### PR TITLE
Cooking appliance boards, cooking and other machine fixes

### LINKED ISSUE #7172: Ovens preheat horrifically slowly and lose heat rapidly, deep fryers only reach 87.6% efficiency at maximum temperature
Following the merge of #7164 

4WdCBXn
Fryer at maximum overdrive. The oven itself reaches 100% efficiency when it reaches 200°C.

Ovens function as they should now, they just take a million years to cook anything for the first half an hour after you show up, or for half an hour after you accidentally walk away for a minute with the door open. This seems like it would be bad for business. Neither one could be upgraded at all, as a side note, but that might be more user error than anything.

### LINKED ISSUE #7190: Ovens seem to not cook until they reach at least 40% efficiency (80°C), candy machines and cereal makers don't work at all
@JohnWildkins 

Whatever you put into an oven before this point will stay at the "It is cold" phase, and anything made in the candy machine or cereal maker will stay at that point for an indefinite amount of time.

### PR DESCRIPTION
**Pull request by JohnWildkins**
Resolves #7172. Resolves #7190.

Cooking appliances are no longer esoteric technology beyond the control of NT's finest researchers.

- Adds microwave, oven, deep fryer, cereal maker, and candy maker circuit boards for research and construction.
- Constructing, upgrading, and deconstructing microwaves and other cooking appliances is now possible. (Note, as of yet upgrading microwaves has no purpose. Will be fixed ASAP along w/ microwave overhaul)
- Oven efficiency now starts at 70% once pre-heated, to alleviate the incredibly slow cooking times at round-start.
- Oven heat loss from the door being open cut down to 1.5x from 4x.
- Deconstructing a machine now returns cable coil.
- Wirecutting a blueprint on the cable-coil stage lowers it to the 'secured blueprint' stage, requiring a wrench to disassemble instead of wirecutters again.
- Printing a smart-fridge board no longer gives you an operating table circuit board.
- Deconstructing a machine no longer displays the incorrect icon state (box instead of blueprint).
- Candy and cereal makers work again. (woops)

### CODE REVIEW

**Comment by JohnWildkins** [comment]
!review

File: code/game/machinery/constructable_frame.dm [hunk 1]
```diff
@@ -110,7 +110,7 @@
 					if(P.iswirecutter())
 						playsound(src.loc, P.usesound, 50, 1, pitch_toggle)
 						to_chat(user, span("notice", "You remove the cables."))
-						state = 1
+						state = 2
```
Comment (NonQueueingMatt) [inline code/game/machinery/constructable_frame.dm]: I hate these magic numbers.

File: code/game/machinery/kitchen/cooking_machines/_appliance.dm [hunk 2]
```diff
@@ -214,6 +207,8 @@
 	else if(istype(check, /obj/item/weapon/disk/nuclear))
 		to_chat(user, "<span class='warning'>You can't cook that.</span>")
 		return 0
+	else if(istype(I, /obj/item/weapon/crowbar) || istype(I, /obj/item/weapon/screwdriver) || istype(I, /obj/item/weapon/storage/part_replacer))
+		return 0
```
Comment (NonQueueingMatt) [inline code/game/machinery/kitchen/cooking_machines/_appliance.dm]:
```suggestion
		return FALSE
```
For all of these, if it's meant to be false/true instead of a number.

File: code/game/machinery/kitchen/cooking_machines/oven.dm [hunk 3]
```diff
@@ -72,17 +79,25 @@
 
 	if (open)
 		open = 0
-		loss = (active_power_usage / resistance)*0.5
+		loss = (heating_power / resistance) * 0.5
 	else
 		open = 1
```
Comment (NonQueueingMatt) [inline code/game/machinery/kitchen/cooking_machines/oven.dm]:
```suggestion
		open = TRUE
```

File: code/game/machinery/kitchen/cooking_machines/oven.dm [hunk 4]
```diff
@@ -72,17 +79,25 @@
 
 	if (open)
 		open = 0
```
Comment (NonQueueingMatt) [inline code/game/machinery/kitchen/cooking_machines/oven.dm]:
```suggestion
		open = FALSE
```

File: code/game/machinery/constructable_frame.dm (hunk 1 above)
Comment (JohnWildkins) [reply code/game/machinery/constructable_frame.dm]: Me too, friend, me too.

File: code/game/machinery/kitchen/cooking_machines/_appliance.dm (hunk 2 above)
Comment (JohnWildkins) [reply code/game/machinery/kitchen/cooking_machines/_appliance.dm]: Done

File: code/game/machinery/kitchen/cooking_machines/_appliance.dm [hunk 5]
```diff
@@ -39,14 +39,7 @@
...
 
-	var/combine_first = 0//If 1, this appliance will do combinaiton cooking before checking recipes
-
-	component_types = list(
-			/obj/item/weapon/circuitboard/cooking,
-			/obj/item/weapon/stock_parts/capacitor = 3,
-			/obj/item/weapon/stock_parts/scanning_module,
-			/obj/item/weapon/stock_parts/matter_bin = 2
-		)
+	var/combine_first = FALSE//If 1, this appliance will do combinaiton cooking before checking recipes
```
Comment (Geevies) [inline code/game/machinery/kitchen/cooking_machines/_appliance.dm]: combinaiton

Comment (JohnWildkins) [reply code/game/machinery/kitchen/cooking_machines/_appliance.dm]: i didnt do it papa
